tor_archivist/core/blossom.py

```diff
@@ -0,0 +1,74 @@
+import requests
+
+class ConfigurationError(Exception):
+    pass
+
+class BlossomAPI(object):
+    def __init__(
+            self,
+            email: str=None,
+            password: str=None,
+            api_key: str=None,
+            api_base_url: str="http://api.grafeas.localhost:8000",
+            login_url: str="http://grafeas.localhost:8000/login/"
+    ):
+        if not email:
+            raise ConfigurationError("Need an email address!")
+        if not password:
+            raise ConfigurationError("Need a password!")
+        if not api_key:
+            raise ConfigurationError("Need an API key!")
+
+        self.email = email
```

Comment:
You might be able to make the validations above a one-liner with `self.email = email if email else raise ValueError('Need an email address')`

Reply:
@itsthejoker thoughts?